### ISSUE TITLE
Connection aborted

### ISSUE THREAD

**Issue by Drovosek01** [ckamte/adobe-packager-windows#5]
I try download After Effects 2024 v24.5.0.052 ru_RU
```
PS D:\TEMP\Adobe\adobe-packager-windows> python.exe .\ccdl-win.py
=================================
=  Adobe CC Package Downloader  =
============= 1.0.0 =============

Available URL versions
[v4]          URL version 4
[v5]          URL version 5
[v6]          URL version 6

Please enter the URL version(v4/v5/v6) for downloading products.xml, or nothing for v6:

Using URL version 6

Available Platforms
[win32]       32 Bit Windows
[win64]       64 Bit Windows
[winarm64]    Windows ARM

Please enter application platform (eg win32), or nothing for win64:

Getting win64 products

Downloading product data

42 products found:
[AEFT]      After Effects
[AEFTBETA]  After Effects (Beta)
[AERO]      Aero (Beta)
[AICY]      InCopy
[AME]       Media Encoder
[AMEBETA]   Media Encoder (Beta)
[APRO]      Acrobat
[AUDT]      Audition
[AUDTBETA]  Audition (Beta)
[CHAR]      Character Animator
[CHARBETA]  Character Animator (Beta)
[DRWV]      Dreamweaver
[ESHR]      Dimension
[FLPR]      Animate
[FRSC]      Fresco
[IDSN]      InDesign
[IDSNBETA]  InDesign (Beta)
[ILST]      Illustrator
[ILSTBETA]  Illustrator Beta
[ILSTPR]    Illustrator (Prerelease)
[KBRG]      Bridge
[KBRGBETA]  Bridge (Beta)
[LRCC]      Lightroom
[LTRM]      Lightroom Classic
[MUSE]      Muse CC
[PHSP]      Photoshop
[PHSPBETA]  Photoshop (Beta)
[PPRO]      Premiere Pro
[PPROBETA]  Premiere Pro (Beta)
[PRLD]      Prelude
[PSXUWP]    Photoshop Express
[RUSH]      Premiere Rush
[RUSHBETA]  Premiere Rush (Beta)
[SBSTA]     Substance 3D Sampler
[SBSTABETA] Substance 3D Sampler (Beta)
[SBSTD]     Substance 3D Designer
[SBSTP]     Substance 3D Painter
[SHPR]      Substance 3D Modeler
[SHPRBETA]  Substance 3D Modeler (Beta)
[STGR]      Substance 3D Stager
[STGRBETA]  Substance 3D Stager (Beta)
[UXPD]      UXP Developer Tools

Please enter the SAP Code of the desired product from the list above: AEFT

After Effects Platform: win64 - 23.0.0.59
After Effects Platform: win64 - 23.1.0.83
After Effects Platform: win64 - 23.2.0.65
After Effects Platform: win64 - 23.2.1.3
After Effects Platform: win64 - 23.3.0.53
After Effects Platform: win64 - 23.4.0.53
After Effects Platform: win64 - 23.5.0.52
After Effects Platform: win64 - 23.6.0.62
After Effects Platform: win64 - 23.6.2.5
After Effects Platform: win64 - 23.6.5.4
After Effects Platform: win64 - 23.6.6.2
After Effects Platform: win64 - 24.0.0.55
After Effects Platform: win64 - 24.0.1.2
After Effects Platform: win64 - 24.0.2.3
After Effects Platform: win64 - 24.0.3.2
After Effects Platform: win64 - 24.1.0.78
After Effects Platform: win64 - 24.2.1.002
After Effects Platform: win64 - 24.3.0.050
After Effects Platform: win64 - 24.4.0.047
After Effects Platform: win64 - 24.4.1.002
After Effects Platform: win64 - 24.5.0.052

Please enter the desired version. Nothing for 24.5.0.052:

Prepare to download Adobe After Effects, version 24.5.0.052

Available languages: All, de_DE, en_AE, en_GB, en_IL, en_US, es_ES, es_MX, fr_CA, fr_FR, fr_MA, it_IT, ja_JP, ko_KR, pt_BR, ru_RU, zh_CN

Please enter the desired install language, or nothing for [ru_RU]:

Prepare to download Adobe After Effects-24.5.0.052-ru_RU-win64
Downloading product icons

100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.66k/1.66k [00:00<00:00, 883kiB/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.69k/1.69k [00:00<00:00, 1.69MiB/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 915/915 [00:00<00:00, 69.3kiB/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 871/871 [00:00<00:00, 578kiB/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3.31k/3.31k [00:00<00:00, 1.65MiB/s]
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 481/481 [00:00<00:00, 319kiB/s]

Downloading main app package

Selected 1 core packages and 1 non-core packages

Creating Application.json

[AEFT_24.5.0.052] Downloading AdobeAfterEffects24AllTrial.zip
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.42G/1.42G [02:47<00:00, 8.43MiB/s]
[AEFT_24.5.0.052] Downloading AdobeAfterEffects24AllTrial_maxon.zip
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 344M/344M [00:32<00:00, 10.6MiB/s]

Downloading dependency packages
Traceback (most recent call last):
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 789, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 536, in _make_request
    response = conn.getresponse()
               ^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connection.py", line 464, in getresponse
    httplib_response = super().getresponse()
                       ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1428, in getresponse
    response.begin()
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 331, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 300, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 843, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\retry.py", line 474, in increment
    raise reraise(type(error), error, _stacktrace)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\util\util.py", line 38, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 789, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connectionpool.py", line 536, in _make_request
    response = conn.getresponse()
               ^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\urllib3\connection.py", line 464, in getresponse
    httplib_response = super().getresponse()
                       ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1428, in getresponse
    response.begin()
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 331, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 300, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\TEMP\Adobe\adobe-packager-windows\ccdl-win.py", line 1035, in <module>
    run_ccdl(products, cdn, sapCodes, selectedPlatform)
  File "D:\TEMP\Adobe\adobe-packager-windows\ccdl-win.py", line 957, in run_ccdl
    dependencies_download(
  File "D:\TEMP\Adobe\adobe-packager-windows\ccdl-win.py", line 850, in dependencies_download
    dep_json = get_application_json(firstItem["buildGuid"])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\TEMP\Adobe\adobe-packager-windows\ccdl-win.py", line 595, in get_application_json
    response = session.get(ADOBE_APPLICATION_JSON_URL, headers=headers)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 602, in get
    return self.request("GET", url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python312\Lib\site-packages\requests\adapters.py", line 682, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
```
But I got error `requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))`

I am on Windows 10 22H2 x64

**Comment by ckamte** [comment]
Hi @Drovosek01 

It's look like server problem.

```
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='prod-rel-ffc-ccm.oobesaas.adobe.com', port=443): Max retries exceeded with url: /adobe-ffc-external/core/v6/products/all?_type=xml&channel=ccm&channel=sti&platform=win64,win32&productType=Desktop (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000021B24AB5AF0>: Failed to establish a new connection: [WinError 10013] An attempt was made to access a socket in a way forbidden by its access permissions'))
```

**Comment by ckamte** [comment]
I also test with language ru_RU and its OK too.

```
C:\Adobe_Download\2024\Media>python ccdl_win.py -u 6 -p win64 -l ru_RU -x
=================================
=  Adobe CC Package Downloader  =
============= 1.0.0 =============


Using URL version 6


Using provided version: win64

Getting win64 products

Downloading product data

42 products found:
[AEFT]      After Effects
[AEFTBETA]  After Effects (Beta)
[AERO]      Aero (Beta)
[AICY]      InCopy
[AME]       Media Encoder
[AMEBETA]   Media Encoder (Beta)
[APRO]      Acrobat
[AUDT]      Audition
[AUDTBETA]  Audition (Beta)
[CHAR]      Character Animator
[CHARBETA]  Character Animator (Beta)
[DRWV]      Dreamweaver
[ESHR]      Dimension
[FLPR]      Animate
[FRSC]      Fresco
[IDSN]      InDesign
[IDSNBETA]  InDesign (Beta)
[ILST]      Illustrator
[ILSTBETA]  Illustrator Beta
[ILSTPR]    Illustrator (Prerelease)
[KBRG]      Bridge
[KBRGBETA]  Bridge (Beta)
[LRCC]      Lightroom
[LTRM]      Lightroom Classic
[MUSE]      Muse CC
[PHSP]      Photoshop
[PHSPBETA]  Photoshop (Beta)
[PPRO]      Premiere Pro
[PPROBETA]  Premiere Pro (Beta)
[PRLD]      Prelude
[PSXUWP]    Photoshop Express
[RUSH]      Premiere Rush
[RUSHBETA]  Premiere Rush (Beta)
[SBSTA]     Substance 3D Sampler
[SBSTABETA] Substance 3D Sampler (Beta)
[SBSTD]     Substance 3D Designer
[SBSTP]     Substance 3D Painter
[SHPR]      Substance 3D Modeler
[SHPRBETA]  Substance 3D Modeler (Beta)
[STGR]      Substance 3D Stager
[STGRBETA]  Substance 3D Stager (Beta)
[UXPD]      UXP Developer Tools

Please enter the SAP Code of the desired product from the list above: aeft

After Effects Platform: win64 - 23.0.0.59
After Effects Platform: win64 - 23.1.0.83
After Effects Platform: win64 - 23.2.0.65
After Effects Platform: win64 - 23.2.1.3
After Effects Platform: win64 - 23.3.0.53
After Effects Platform: win64 - 23.4.0.53
After Effects Platform: win64 - 23.5.0.52
After Effects Platform: win64 - 23.6.0.62
After Effects Platform: win64 - 23.6.2.5
After Effects Platform: win64 - 23.6.5.4
After Effects Platform: win64 - 23.6.6.2
After Effects Platform: win64 - 24.0.0.55
After Effects Platform: win64 - 24.0.1.2
After Effects Platform: win64 - 24.0.2.3
After Effects Platform: win64 - 24.0.3.2
After Effects Platform: win64 - 24.1.0.78
After Effects Platform: win64 - 24.2.1.002
After Effects Platform: win64 - 24.3.0.050
After Effects Platform: win64 - 24.4.0.047
After Effects Platform: win64 - 24.4.1.002
After Effects Platform: win64 - 24.5.0.052

Please enter the desired version. Nothing for 24.5.0.052:

Prepare to download Adobe After Effects, version 24.5.0.052

Using provided language: ru_RU

Prepare to download Adobe After Effects-24.5.0.052-ru_RU-win64
Downloading product icons

Downloaded file is OK, skipping ...

Downloaded file is OK, skipping ...

Downloaded file is OK, skipping ...

Downloaded file is OK, skipping ...

Downloaded file is OK, skipping ...

Downloaded file is OK, skipping ...


Downloading main app package

Selected 1 core packages and 1 non-core packages

Creating Application.json

[AEFT_24.5.0.052] Downloading AdobeAfterEffects24AllTrial.zip
Downloaded file is OK, skipping ...

[AEFT_24.5.0.052] Downloading AdobeAfterEffects24AllTrial_maxon.zip
Downloaded file is OK, skipping ...


Downloading dependency packages

Selected 0 core packages and 2 non-core packages

Creating Application.json

[COMP_1.4.1] Downloading AdobeColor_MotionPicture_1_0-mul.zip
Downloaded file is OK, skipping ...

[COMP_1.4.1] Downloading AdobeColor_VideoProfilesAE_1_0-mul.zip
Downloaded file is OK, skipping ...


Selected 0 core packages and 1 non-core packages

Creating Application.json

[CORG_1.1.4] Downloading AdobeColorCommonSetRGB_1_0-mul.zip
Downloaded file is OK, skipping ...


Selected 3 core packages and 0 non-core packages

Creating Application.json

[ACR_16.4.0.30] Downloading CameraRawRIBSCoExistPackage.zip
Downloaded file is OK, skipping ...

[ACR_16.4.0.30] Downloading AdobeCameraRaw8.0All-x64.zip
Downloaded file is OK, skipping ...

[ACR_16.4.0.30] Downloading AdobeCameraRawProfile8.0All.zip
Downloaded file is OK, skipping ...


Selected 6 core packages and 5 non-core packages

Creating Application.json

[CCXP_5.5.2.5] Downloading CCXProcess-LaunchAgent.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading CCX-Process-mul-x64.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading CCXProcess-Components.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.start-2.7.2.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.start-2.7.2-manifest.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.start-2.16.0.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.start-2.16.0-manifest.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.fnft-3.5.0.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.fnft-3.5.0-manifest.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.start-7.7.0.zip
Downloaded file is OK, skipping ...

[CCXP_5.5.2.5] Downloading com.adobe.ccx.start-7.7.0-manifest.zip
Downloaded file is OK, skipping ...

Generating Driver.xml

Successfully downloaded Adobe After Effects v-24.5.0.052


Do you want to download another package (Y/n):
```

**Comment by ckamte** [comment]
Check this link

[https://stackoverflow.com/questions/67809815/remote-end-closed-connection-without-response/67809908#67809908](url)

But it think headers are fine.

OR

Turn off your firewall and test again.
